### PR TITLE
Refactor editor to use Redux

### PR DESCRIPTION
**Pull request by ICimpoes**
Use Redux to store and modify canvas elements

### CODE REVIEW

File: src/Palette.tsx [hunk 1]
```diff
@@ -1,46 +1,47 @@
...
     );
 }
 
 interface ShapeProps {
-    shape: ShapeType;
-    onDragStart: (shape: ShapeType) => void;
+    shapeType: ShapeType;
 }
 
 function Shape(props: ShapeProps) {
```
Comment (ardevelop) [inline src/Palette.tsx]: It's nice when there is one module per component

File: src/Shapes.tsx [hunk 2]
```diff
@@ -1,76 +1,58 @@
 import React from "react";
 import { v4 as uuidv4 } from "uuid";
+import { Point } from "./models";
 
 export const Shapes = {
     circle: (props: shapeProps) => {
```
Comment (ardevelop) [inline src/Shapes.tsx]:
```tsx
export const Shapes = {
  circle: Circle,
  square: Square,
  triangle: Triangle
}
```

and use:

```tsx
const Shape = Shapes[props.shapeType];

<Shape point={{x: 25, y: 25}}>
```

File: src/Workspace.tsx [hunk 3]
```diff
@@ -1,52 +1,29 @@
...
-            if (dragElement !== undefined) {
-                const elements = canvasElements.slice();
-                elements[dragElement].point = {
-                    x: e.nativeEvent.offsetX,
-                    y: e.nativeEvent.offsetY,
-                };
-                setCanvasElements(elements);
-            }
+            dispatch(
+                move({ x: e.nativeEvent.offsetX, y: e.nativeEvent.offsetY })
```
Comment (ardevelop) [inline src/Workspace.tsx]: We can extract repeating logic into a helper function

File: src/Shapes.tsx (hunk 2 above)
Comment (ICimpoes) [reply src/Shapes.tsx]: Updated, looks much better.

**Review by ardevelop — APPROVED** [review]

**Pull request merged**